src/System.Net.Sockets/tests/FunctionalTests/Shutdown.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using Xunit.Abstractions;
+
+using System.Threading.Tasks;
+
+namespace System.Net.Sockets.Tests
+{
+    public class Shutdown
+    {
+        private readonly ITestOutputHelper _log;
+
+        public Shutdown(ITestOutputHelper output)
+        {
+            _log = output;
+        }
+
+        private static void OnOperationCompleted(object sender, SocketAsyncEventArgs args)
+        {
+            Assert.Equal(SocketError.Success, args.SocketError);
+
+            switch (args.LastOperation)
+            {
+                case SocketAsyncOperation.Accept:
+                {
+                    Socket client = args.AcceptSocket;
+                    args.SetBuffer(new byte[1], 0, 1);
+                    args.UserToken = client;
+
+                    Assert.True(client.ReceiveAsync(args));
+                    break;
+                }
+
+                case SocketAsyncOperation.Receive:
+                {
+                    var client = (Socket)args.UserToken;
+                    if (args.BytesTransferred == 0)
+                    {
+                        client.Dispose();
+                        break;
+                    }
+
+                    Assert.True(client.SendAsync(args));
+                    break;
+                }
+
+                case SocketAsyncOperation.Send:
+                {
+                    var client = (Socket)args.UserToken;
+
+                    Assert.True(args.BytesTransferred == args.Buffer.Length);
+                    Assert.True(client.ReceiveAsync(args));
+                    break;
+                }
+            }
+        }
+
+        [Fact]
+        public void Shutdown_TCP_CLOSED_Success()
+        {
+            using (Socket server = new Socket(SocketType.Stream, ProtocolType.Tcp))
+            using (Socket client = new Socket(SocketType.Stream, ProtocolType.Tcp))
+            {
+                int port = server.Bind(IPAddress.IPv6Any);
```

Comment:
> Since we have no documentation, we'll point our customers to the tests

Can you clarify this? Don't we have documentation on MSDN?

> If I read the line it looks like that Bind is actually a Socket member, which is not

Yes, this is a problem with extension methods. I think the general consensus, however, is that the readability gains are worth the potential confusion. That said, I'll rename this to `BindAndAssignPort`.